src/Form/EventListener/CoordinatorManagedAreaListener.php

```diff
@@ -66,10 +66,10 @@ public function onPostSubmit(FormEvent $event): void
             return;
         }
 
-        $adherent->setCoordinatorManagedAreas(
-            $adherent->getCoordinatorManagedAreas()->filter(function (CoordinatorManagedArea $entity) {
-                return !empty($entity->getCodes()) && !empty($entity->getSector());
-            })
-        );
+        foreach ($adherent->getCoordinatorManagedAreas() as $area) {
```

Comment:
Let's move this to `Adherent` class instead. 
This listener would be lighter, it's easier to unit test and more reusable.
What do you think ?

Reply:
With the right method name, it would also bring more semantic

Reply:
@B-Galati why not.
The name like `removeEmptyCoordinatorManagedAreas` ?